viewer/packages/tools/src/Measurement/MeasurementTool.ts

```diff
@@ -0,0 +1,215 @@
+ * required point to get measurement of the distance.
+ * The tools default measurement is in "Meters" as supported in Reveal, but it also provides
+ * user to customise the measuring units based on their convinience with the callback.
+ *
+ * @example
+ * ```js
+ * const measurementTool = new MeasurementTool(viewer);
+ * measurementTool.enterMeasurementMode();
+ * // ...
+ * measurementTool.exitMeasurementMode();
+ *
+ * // detach the tool from the viewer
+ * measurementTool.dispose();
+ * ```
+ * @example
+ * ```jsx runnable
+ * const measurementTool = new MeasurementTool(viewer, {changeMeasurementLabelMetrics: (distance) => {
+ *    // 1 meters = 3.281 feet
+ *    const distanceInFeet = distance * 3.281;
+ *    return { distance: distanceInFeet, units: 'ft'};
+ *  }});
+ *  measurementTool.enterMeasurementMode();
+```
+ */
+export class MeasurementTool extends Cognite3DViewerToolBase {
+  private _options: Required<MeasurementOptions>;
+
+  private readonly _viewer: Cognite3DViewer;
+  private readonly _geometryGroup = new THREE.Group();
+  private readonly _measurements: Measurement[];
+  private _currentMeasurementIndex: number;
+  private readonly _htmlOverlay: HtmlOverlayTool;
+
+  private readonly _handleLabelClustering = this.createCombineClusterElement.bind(this);
+  private readonly _handlePointerClick = this.onPointerClick.bind(this);
+  private readonly _handlePointerMove = this.onPointerMove.bind(this);
+
+  private readonly _overlayOptions: HtmlOverlayToolOptions = {
+    clusteringOptions: { mode: 'overlapInScreenSpace', createClusterElementCallback: this._handleLabelClustering }
+  };
+
+  private static readonly defaultLineOptions: Required<MeasurementOptions> = {
+    distanceToLabelCallback: d => MeasurementTool.metersLabelCallback(d),
+    lineWidth: 2.0,
+    color: 0x00ffff
+  };
+
+  constructor(viewer: Cognite3DViewer, options?: MeasurementOptions) {
+    super();
+    this._viewer = viewer;
+    this._options = {
+      ...MeasurementTool.defaultLineOptions,
+      ...options
+    };
+    this._measurements = [];
+    this._htmlOverlay = new HtmlOverlayTool(this._viewer, this._overlayOptions);
+    this._currentMeasurementIndex = -1;
+
+    this._geometryGroup.name = MeasurementTool.name;
+    this._viewer.addObject3D(this._geometryGroup);
+  }
+
+  /**
+   * Enter into point to point measurement mode.
+   */
+  enterMeasurementMode(): void {
+    this.setupEventHandling();
+  }
+
+  /**
+   * Exit measurement mode.
+   */
+  exitMeasurementMode(): void {
+    //clear all mesh, geometry & event handling.
+    this._measurements.forEach(measurement => {
+      measurement.dispose();
+    });
+    this.removeEventHandling();
+  }
+
+  /**
+   * Removes a measurement from the Cognite3DViewer.
+   * @param _measurement Measurement mesh to be removed from @Cognite3DViewer.
+   */
+  removeMeasurement(_measurement: THREE.Group): void {
+    // TODO 2022-07-05 larsmoa: Implement
+    throw new Error('Not implemented');
+  }
```

Comment:
This must be implemented again, but don't accept group as mentioned in my comments about the docs.